src/roswire/proxy/ros2state.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+__all__ = ('SystemState', 'ROS2SystemStateProbe')
+
+from typing import Dict, List
+
+import attr
+import dockerblade
+
+from state import SystemState
+from .app.instance import AppInstance
+
+
+@attr.s(frozen=True, auto_attribs=True)
+class ROS2SystemStateProbe:
+    """Provides an interface for obtaining the instantaneous state of a ROS
+    system in terms of its publishers, subscribers, and services."""
+    _app_instance: AppInstance = attr.ib()
+
+    def probe(self) -> SystemState:
+        """Obtains the instantaneous state of the associated ROS system."""
+        mode_dict: Dict[str, Dict[str, List[str]]] = {'pub': {},
+                                                      'sub': {},
+                                                      'serv': {}}
+        command = "rosnode list"
+        try:
+            output = self._app_instance.shell.check_output(command, text=True)
+        except dockerblade.exceptions.CalledProcessError as error:
+            raise error
+        node_names = output.split('\r\n')
+        for name in node_names:
+            info = f"ros2 node info '{name}'"
+            mode = 'None'
+            try:
+                output = self._app_instance.shell.check_output(info, text=True)
```

Comment:
The `except` clause should come immediately after the `output = ...` line. In general, try to keep the scope of the `try` block as small as possible. In this case, the error can only occur on the `output = ...` line.